llpc/context/llpcGraphicsContext.cpp

```diff
@@ -263,4 +263,21 @@ ArrayRef<ResourceMappingNode> GraphicsContext::mergeUserDataNodeTable(SmallVecto
 }
 #endif
 
+// =====================================================================================================================
+// Check whether the pipeline uses features relevant to subgroup size
+bool GraphicsContext::useSubgroupSize() const {
+  std::array<const PipelineShaderInfo *, 5> shaderInfos = {
+      &m_pipelineInfo->vs, &m_pipelineInfo->tcs, &m_pipelineInfo->tes, &m_pipelineInfo->gs, &m_pipelineInfo->fs,
+  };
+  for (uint32_t i = 0; i < shaderInfos.size(); ++i) {
+    if (!shaderInfos[i]->pModuleData)
+      continue;
+    const ShaderModuleData *moduleData = reinterpret_cast<const ShaderModuleData *>(shaderInfos[i]->pModuleData);
```

Comment:
nit: you can use `auto* moduleData = ...`. The return type is obvious because it's on the right hand side.